backend/src/lang/intents/handlers/handler.enddateopen.ts

```diff
@@ -0,0 +1,30 @@
+import { EmploymentContractService } from './../../../database/employmentContract/employmentContract.service';
+import { Injectable } from '@nestjs/common';
+import { IntentHandler } from './handler.abstract';
+
+@Injectable()
+export class EndDateOpenIntentHandler extends IntentHandler{
+
+    constructor(private employmentContractService: EmploymentContractService){
+        super();
+    }
+
+    /**
+     * Proccesses a given DialogFlow Reponse
+     * @param intentData Parts of the DialogFlow response
+     * @returns {Promise<ReturnText | undefined>} The text for the user or undefined.
+     */
+    public async handle(intentData: IIntentData): Promise<ReturnText | undefined> {
+        try {
+            const contractName = this.extractData(intentData.parameter, 'EmploymentContract', 'EmploymentContract');
+
+            // If our parameters are not ready Dialogflow will ask for them
+            if (contractName !== '') {
+                await this.employmentContractService.editEndDateString(contractName, 'unbefristet');
+            }
+            return undefined;
+          } catch (error) {
+            return { text: 'Beim Ändern des Enddatums ist ein Fehler aufgetreten. Bitte versuche es erneut' };
```

Comment:
Bei EndDateIntentHandler findet kein Catch in der gleichen Situation (Schreiben eines Datenbankupdates) statt. Ich persönlich finde diese Lösung hier natürlich schöner, aber auch allgemein sollten die beiden Handler hier vielleicht zumindest aneinander angeglichen werden.